lib/peep.rb

```diff
@@ -0,0 +1,47 @@
+require 'date'
+require 'pg'
+
+class Peep
+
+  attr_reader :content, :user_id, :time_posted
+
+  def initialize(time_posted, user_id, content)
+   
+    @time_posted = time_posted
+    @user_id = user_id
+    @content = content
+  
+  end
+
+  def self.all
+
+    @peeps = []
+
+    if ENV['RACK_ENV'] == 'test'
+      con = PG.connect :dbname => 'chitter_test'
+    else
+      con = PG.connect :dbname => 'chitter'
+    end    
+    peeps = con.exec "SELECT * FROM peeps"
+    peeps.each do |row|
+      @peeps << Peep.new(row['time_posted'],row['user_id'],row['content'])
```

Comment:
Could use `peeps.map do |row| #rest of code` to iterate and change values within itself rather than creating a separate array.